frontend/src/app/modules/members/invite/invite.component.ts

```diff
@@ -0,0 +1,156 @@
+import { debounceTime, switchMap } from 'rxjs/operators';
+import { RoleService } from '@core/services/role.service';
+import { Role } from '@shared/models/role/role';
+import { TeamOption } from '@shared/models/teams/team-option';
+import { NewMember } from '@shared/models/member/new-member';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { AuthenticationService } from '@core/services/authentication.service';
+import { User } from '@shared/models/user/user';
+import { Organization } from '@shared/models/organization/organization';
+import { ToastNotificationService } from '@core/services/toast-notification.service';
+import { TeamService } from '@core/services/team.service';
+import { UserService } from '@core/services/user.service';
+import { ShareDataService } from '@core/services/share-data.service';
+import { Invition } from '@shared/models/member/invition';
+
+@Component({
+    selector: 'app-invite',
+    templateUrl: './invite.component.html',
+    styleUrls: ['./invite.component.sass']
+})
+export class InviteComponent extends BaseComponent implements OnInit {
+    @Output() closeModal = new EventEmitter<void>();
+    notMembers: User[];
+    searchTerm: Subject<string> = new Subject<string>();
+    roles: Role[];
+    teams: TeamOption[];
+
+    loadingNumber: number = 0;
+
+    user: User;
+    organization: Organization;
+
+    invations: Invition[] = [{ member: {} as NewMember, groupForm: this.generateGroupForm() }];
+    constructor(
+        private memberService: MemberService,
+        private roleService: RoleService,
+        private authService: AuthenticationService,
+        private toastNotifications: ToastNotificationService,
+        private teamService: TeamService,
+        private userService: UserService,
+        private updateDataService: ShareDataService<Member>
+    ) {
+        super();
+    }
+
+    ngOnInit(): void {
+        this.loadData();
+        this.user = this.authService.getUser();
+    }
+
+    generateGroupForm() {
+        return new FormGroup({
+            name: new FormControl(
+                '',
+                [
+                    Validators.required,
+                ]
+            ),
+            team: new FormControl(
+                []
+            ),
+            role: new FormControl(
+                '',
+                [
+                    Validators.required,
+                ]
+            )
+        });
+    }
+
+    loadData() {
+        this.searchTerm.pipe(
+            this.untilThis,
+            debounceTime(300),
+            switchMap((term: string) => {
+                this.loadingNumber += 1;
+                return this.userService.searchMembersNotInOrganization(this.organization.id, term)
+                    .pipe(this.untilThis);
+            })
+        ).subscribe(users => {
+            this.notMembers = users
+                .filter(u => !this.invations.some(i => i.groupForm.value.name.id === u.id))
```

Comment:
It would be better to do it on the backend.
Now it looks like you load all users and after that filter them in the frontend. Please correct me if I'm wrong

Reply:
We have filtering by email on backend, and on frontend we should filter users selected in other fields. Example on screen.
![image](https://user-images.githubusercontent.com/75535425/129494137-b7619ffa-903e-4a41-b65e-8ea934c09f62.png)
We have selected in first field a arnoldo_schiller@hotmail.com and we can't select it in second fiels.